developer-extension/src/panel/components/tabs/eventsTab/computeFacetState.ts

```diff
@@ -0,0 +1,59 @@
+import type { FacetRegistry, FacetValuesFilter } from '../../../hooks/useEvents'
+import type { Facet, FacetValue } from '../../../facets.constants'
+type SelectionState = 'selected' | 'unselected' | 'partial-selected'
+
+export function computeSelectionState(
+  facetValuesFilter: FacetValuesFilter,
+  facetRegistry: FacetRegistry,
+  facet: Facet,
+  facetValue: FacetValue,
+  parentList: string[]
+): SelectionState {
+  const childrenFacets = facet.values?.[facetValue]?.facets
+
+  // we cannot know how many children in total there are, so we need to have facetRegistry
+  const children =
+    childrenFacets && childrenFacets.flatMap((child: Facet) => facetRegistry.getFacetChildrenValues(child.path))
+  const filteredFacetValues = Object.values(facetValuesFilter.facetValues).flat()
+  const isFiltering = !!Object.keys(facetValuesFilter.facetValues)
+
+  if (facetValuesFilter.type === 'include') {
+    if (!isFiltering) {
+      return 'unselected'
+    }
+
+    for (const parent of parentList) {
+      if (filteredFacetValues.includes(parent)) {
+        return 'selected'
+      }
+    }
+
+    // if all children are in the filter, then it should be selected'
+    if (children && children.every((child: FacetValue) => filteredFacetValues.includes(child))) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I would extract duplicates `children.every` and `children.some`  before the `type` condition and reuse the results in both cases.